File: 190527/sony/exam.py
def solution(answers):
    s1 = [1,2,3,4,5]  # cycle : 5
    s2 = [2,1,2,3,2,4,2,5]  # cycle : 8
    s3 = [3,3,1,1,2,2,4,4,5,5]  # cycle : 10

    a1 = 0
    a2 = 0
    a3 = 0

    for i in range(len(answers)):
        if(s1[i%len(s1)] == answers[i]):
            a1 +=1
        if(s2[i%len(s2)] == answers[i]):
            a2 +=1
        if(s3[i%len(s3)] == answers[i]):
            a3 +=1

    answer_list = [a1,a2,a3]
    m = max(answer_list)
    result = []
    for i in range(len(answer_list)):
        if m == answer_list[i]:
            result.append(i+1)
    return result

File: 190527/circus/exam.js
function solution(answers) {
    var num1 = [1, 2, 3, 4, 5];
    var num2 = [2, 1, 2, 3, 2, 4, 2, 5];
    var num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    var results = [0, 0, 0];
    var winer = [];

    answers.forEach((ans, i) => {
        if (ans === num1[i % num1.length]) results[0] += 1;
        if (ans === num2[i % num2.length]) results[1] += 1;
        if (ans === num3[i % num3.length]) results[2] += 1;
    })

    var max = Math.max(...results);
    results.forEach((result, i) => {
        if (result === max) winer.push(i + 1);
    })
    return winer;
}